src/colmap/controllers/incremental_mapper.cc

```diff
@@ -253,6 +267,86 @@ bool IncrementalPipeline::LoadDatabase() {
     return false;
   }
 
+  // If prior positions are to be used and setup from the database, convert
+  // geographic coords. to cartesian ones
+  if (options_->use_prior_position &&
+      options_->set_prior_position_from_database) {
+    return SetupPriorPoseFromDatabase();
+  }
+
+  return true;
+}
+
+bool IncrementalPipeline::SetupPriorPoseFromDatabase() {
+  LOG(INFO) << "Setting up prior positions";
+
+  Timer timer;
+  timer.Start();
+
+  bool prior_is_gps = true;
+
+  // GPS reference to be used for EllToENU conversion
+  double ref_lat = std::numeric_limits<double>::max();
+  double ref_lon = std::numeric_limits<double>::max();
+
+  // Get ordered image_ids to use image with lowest ID as the GPS ref.
+  std::set<image_t> image_ids_with_prior;
+  for (const auto& image : database_cache_->Images()) {
+    if (image.second.HasPosePrior()) {
+      image_ids_with_prior.insert(image.first);
+    }
+  }
```

Comment:
Images in the `DatabaseCache` are not ordered so you need to explicitly sort them as in https://github.com/colmap/colmap/blob/bb6a67990fd069baddac56d3e4604ee10d658b29/src/colmap/feature/matcher.cc#L127-L130

Reply:
Yes that's why I have used a set instead of a vector here, so that the image_ids are automatically sorted.  Would you prefer that I use a vector and sort it afterward?

Reply:
Sorry, I had missed this. It's probably fine.